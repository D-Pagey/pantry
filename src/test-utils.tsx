import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
