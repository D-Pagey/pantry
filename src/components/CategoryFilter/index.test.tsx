import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../test-utils';
import { CategoryFilter } from '.';

const props = {
    selected: 'all',
    setSelected: () => null
};

describe('CategoryFilter component', () => {
    it('should render', () => {
        const { container } = render(<CategoryFilter {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should handle click', () => {
        const setSelected = jest.fn();

        const { getByText } = render(<CategoryFilter {...props} setSelected={setSelected} />);

        userEvent.click(getByText('Fruit'));

        expect(setSelected).toHaveBeenCalledWith('fruit');
    });
});
