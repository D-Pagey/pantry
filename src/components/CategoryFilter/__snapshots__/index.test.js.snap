// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFilter component should render 1`] = `
.c0 {
  background-color: #348841;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  cursor: pointer;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: bold;
  padding: 1rem;
  text-align: center;
}

.c2:hover {
  border-bottom: 2px solid white;
}

<div
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      All
    </li>
    <li
      class="c2"
    >
      Veg
    </li>
    <li
      class="c2"
    >
      Fruit
    </li>
    <li
      class="c2"
    >
      Meat
    </li>
    <li
      class="c2"
    >
      Dairy
    </li>
    <li
      class="c2"
    >
      Fish
    </li>
  </ul>
</div>
`;
